bmrg-flow/templates/configmap.yaml

```diff
@@ -12,7 +12,11 @@ metadata:
 data:
   application.properties: |
     server.port=8080
+    {{- if $.Values.global.database.mongodb.tls.enable }}
+    spring.data.mongodb.uri=mongodb://{{ $.Values.global.database.mongodb.user }}:${MONGODB_PASSWORD}@{{ $.Values.global.database.mongodb.host }}
+    {{- else }}
     spring.data.mongodb.uri=mongodb://{{ $.Values.global.database.mongodb.user }}:${MONGODB_PASSWORD}@{{ $.Values.global.database.mongodb.host }}:{{ $.Values.global.database.mongodb.port }}/boomerang
+    {{- end }}
```

Comment:
Could we not always send the port through?

Reply:
Technically I need another flag that can be used to switch replicaSet enable/disable.  TLS is really just for whether we need to import a certificate into the cacerts keystore.  I will update this.

Reply:
Well we also dont want to add to many flags if we don't need them.

Can you just wrap the : and port with an if empty check don't add. So if you don't want to use, don't provide.

Reply:
Hey @gchickma have you thought about this one. It might be something like this (but I haven't valided)

```
spring.data.mongodb.uri=mongodb://{{ $.Values.global.database.mongodb.user }}:${MONGODB_PASSWORD}@{{ $.Values.global.database.mongodb.host }}{{ if $.Values.global.database.mongodb.port }}:{{ $.Values.global.database.mongodb.port }}{{ end }}/boomerang
```

Reply:
So ive done a big of search of other helm charts. I would say majority of them use the design of split parameters i.e. host / port / database / credentials / extraArgs vs url. I think URL makes it hard when dealing with credentials from secret though because in some parts of the yamls (like the deployment-service) it pulls the credentials from environment variables.

if we can, lets go with one long embedded if statements.

lets add `extraArgs` as a parameter.

Reply:
Updated schema

```
      host: bmrg-mdb001-ibm-mongodb-dev
      port: 27017
      database: boomerang
      args:
      user: boomerang
      password: passw0rd
      secretName:
```